### ISSUE TITLE
Providing an interface to `find`

### ISSUE THREAD

**Issue by stefanct** [geany/geany#1178]
I use the "Find in files" dialog extensively in my everyday work *because* it is a very elegant interface to `grep` but nevertheless integrates so well with Geany. However, sometimes I am not interested in the contents of files but merely there names. In these situations I would love to have a similar interface to POSIX `find` that simply takes find arguments and presents any found paths similarly to the "Find in files" dialog (i.e. clickable to open the respective file). Please consider this as a feature request - I won't try to implement it (partially due to the outcome of my "reload all" PR #54 :P)

**Comment by elextr** [comment]
Does adding `-l` to the extra options do what you want?

If not you could also make a plugin to do what you want, adding a new tab to the message window for the results.

PR #54 seems to be waiting for the OP to do stuff, you should tell them, or at least tell them to reply to the question of "whats the status?" since getting no answer to that was what got it closed ;-P

**Comment by stefanct** [comment]
`-l` is *not* a workaround since the search criteria (available in `find`) are the main point.

Instead of another message tab I would rather envision it as a kind of a a clone of the existing `grep` dialog (i.e. launchable via the edit menu) with the opportunity to enrich it even more like `mate-search-tool` (not required by me at all but this would increase usability for people not familiar with `find`, cf. https://windywindycitytech.files.wordpress.com/2013/01/screenshot-search2bfor2bfiles.png)

In #54 the OP waited for answers to his questions before abandoning it mentally ;)

**Comment by elextr** [comment]
> -l is not a workaround since the search criteria (available in find) are the main point.

Oh, ok, I read your original post as "just getting filenames" was the point, not changing the search expression form.

If you want to provide an _alternative_ FIF search mechanism I would definitely say it should go in a plugin first, you can use the existing message window if you want, `msgwin_msg_add()` is in the plugin interface.  I don't think there is any reason it needs to be in core Geany and that way you can add and test all the features incrementally in your own time.

Plugins can add to the main menus, but its not encouraged since if two plugins try for the same menu there is nothing to mediate who gets which position and a mess can ensue.  Same problem exists for plugins defining default keybindings.  Better to stick to tools menu for now.

Plugins that are found to be universally useful can be promoted to core plugins and distributed with Geany itself, like save-actions, splitwindow etc are.

On #54 they (and this applies to all other PR proposers) should ping it if they were expecting answers and didn't get them in a "reasonable" time, Geany contributors are all doing it in their own time, and they are only human and can miss things, go on holidays, be sidetracked or forgetful ... what were we talking about? :smile:

**Comment by techee** [comment]
@stefanct Not sure if it's what you need but the ProjectOrganizer plugin from geany-plugins allows searching for files based on their name and displays the results in the message window. This however only works for project files.

**Comment by stefanct** [comment]
@techee thanks for the suggestion but `find` supports way more than just searching for patterns of names.

**Comment by techee** [comment]
Yeah, sure. I didn't know what exactly you need and thought simple search might be enough.
